src/imitation/util/logger.py

```diff
@@ -0,0 +1,200 @@
+import contextlib
+import os
+from typing import ContextManager, Optional, Sequence
+
+import stable_baselines.logger as sb_logger
+
+
+def _build_output_formats(folder: str,
+                          format_strs: Sequence[str] = None,
+                          ) -> Sequence[sb_logger.KVWriter]:
+  """Build output formats for initializing a Stable Baselines Logger.
+
+  Args:
+    folder: Path to directory that logs are written to.
+    format_strs: An list of output format strings. For details on available
+      output formats see `stable_baselines.logger.make_output_format`.
+  """
+  os.makedirs(folder, exist_ok=True)
+  output_formats = [sb_logger.make_output_format(f, folder)
+                    for f in format_strs]
+  return output_formats
+
+
+class _AccumulatingLogger(sb_logger.Logger):
+
+  def __init__(self,
+               format_strs: Sequence[str],
+               *,
+               mean_logger: sb_logger.Logger,
+               subdir: str):
+    """Like Logger, except also accumulates logkv_mean on the mean_logger.
+
+    Args:
+      format_strs: An list of output format strings. For details on available
+        output formats see `stable_baselines.logger.make_output_format`.
+      mean_logger: A background logger that is used to keep track of log
+        means.
+      subdir: Used to build the logging directory. Also used in the logging
+        prefix for every key written to both this logger and `mean_logger`.
+    """
+    self.subdir = subdir
+    self.mean_logger = mean_logger
+    folder = os.path.join(self.mean_logger.dir, "raw", subdir)
+    output_formats = _build_output_formats(folder, format_strs)
+    os.makedirs(folder, exist_ok=True)
+    super().__init__(folder, output_formats)
+
+  def logkv(self, key, val):
+    raw_key = os.path.join("raw", self.subdir, key)
```

Comment:
I think I've raised this point before, but don't think we ever got to an agreement, so reopening the discussion.

I still don't understand why we rewrite the keys here. This instance is *only* used for keys of `self.subdir`, so everything logged will be prefixed with `raw/{self.subdir}`. Is this really what we want? The only use case I can see for this is if you wanted to concatenate data from loggers of different subdirs, but we know in general this won't be possible.

Reply:
Maybe the use case you had in mind was the `stdout` output format, where it'll be hard to know which logger things are coming from without prefixing the keys?

Reply:
I think it would be helpful in Tensorboard to be able to have the raw results in a separate plot from the mean results since they have different timescales. (This is the only reason, actually)

Reply:
Haven't done hardcore hp tuning yet, especially with these changes so don't know if this actually important

Reply:
You need to have distinct keys (not just distinct loggers) to group the raw and mean statistics into different plots in Tensorboard